conans/client/tools/files.py

```diff
@@ -103,10 +105,21 @@ def print_progress(_, __):
             pass
 
     with zipfile.ZipFile(filename, "r") as z:
-        if not pattern:
-            zip_info = z.infolist()
-        else:
-            zip_info = [zi for zi in z.infolist() if fnmatch(zi.filename, pattern)]
+        zip_info = z.infolist()
+        if pattern:
+            zip_info = [zi for zi in zip_info if fnmatch(zi.filename, pattern)]
+        if flat_folder:
+            names = [n.replace("\\", "/") for n in z.namelist()]
```

Comment:
Can't we assume that if ``flat_folder`` is given, there should be just 1 element in the root, and it be a directory? No need to check for common_prefix?

Reply:
I think it could be error prone for the user. Because, if you don't check... what do you "flat" exactly? all the first levels in case of several dirs in the root?